central/auth/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,240 @@
+		),
+	)
+
+	s.pool = pgtest.ForT(s.T())
+	s.Require().NotNil(s.pool)
+
+	s.store = pgStore.New(s.pool.DB)
+	s.datastore = New(s.store)
+}
+
+func (s *datastorePostgresTestSuite) TearDownTest() {
+	s.pool.Teardown(s.T())
+	s.pool.Close()
+}
+
+func (s *datastorePostgresTestSuite) TestAddConfig() {
+	testCases := []struct {
+		config *storage.AuthMachineToMachineConfig
+		err    error
+	}{
+		{
+			config: &storage.AuthMachineToMachineConfig{},
+			err:    errox.InvalidArgs,
+		},
+		{
+			config: nil,
+			err:    errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				Id: "some-id",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1h",
+			},
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1s",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "24h1s",
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1h",
+				Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+			},
+			err: errox.InvalidArgs,
+		},
+		{
+			config: &storage.AuthMachineToMachineConfig{
+				TokenExpirationDuration: "1h",
+				Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+				IssuerConfig: &storage.AuthMachineToMachineConfig_Generic{
+					Generic: &storage.AuthMachineToMachineConfig_GenericIssuer{Issuer: "something"}},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		s.Run(fmt.Sprintf("tc %d", i), func() {
+			config, err := s.datastore.AddAuthM2MConfig(s.ctx, tc.config)
+			if tc.err != nil {
+				s.ErrorIs(err, tc.err)
+			} else {
+				s.NoError(err)
+				s.NotEmpty(config.GetId())
+			}
+		})
+	}
+}
+
+func (s *datastorePostgresTestSuite) TestGetConfig() {
+	config, err := s.datastore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		TokenExpirationDuration: "1h",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "something",
+				Role:  "Admin",
+			},
+			{
+				Key:   "aud",
+				Value: "github",
+				Role:  "Continuous Integration",
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	storedConfig, exists, err := s.datastore.GetAuthM2MConfig(s.ctx, config.GetId())
+	s.NoError(err)
+	s.True(exists)
+	s.Equal(config, storedConfig)
+}
+
+func (s *datastorePostgresTestSuite) TestListConfigs() {
+	config1, err := s.datastore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		TokenExpirationDuration: "1h",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "something",
+				Role:  "Admin",
+			},
+			{
+				Key:   "aud",
+				Value: "github",
+				Role:  "Continuous Integration",
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config2, err := s.datastore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
```

Comment:
[nit] `config1` and `config2` look identical attribute-wise except for `id` which will be auto-generated. Could you please make `config2` look slightly differently?